lib/trino-orc/src/main/java/io/trino/orc/OrcReader.java

```diff
@@ -461,8 +461,11 @@ public static ProjectedLayout createProjectedLayout(OrcColumn root, List<List<St
                 return fullyProjectedLayout();
             }
 
-            Map<String, List<List<String>>> dereferencesByField = dereferences.stream().collect(
-                    Collectors.groupingBy(
+            Map<String, List<List<String>>> dereferencesByField = dereferences.stream()
```

Comment:
I think `OrcPageSourceFactory#createOrcPageSource` is a better place to do this instead of doing it repeatedly here - until we resolve https://github.com/trinodb/trino/pull/5575#discussion_r585732651 which also involves comparison with partition metadata. @willmostly @findepi thoughts?

Reply:
My thinking was that since we [downcase the partition metadata](https://github.com/trinodb/trino/blob/b7c52b5649e8a5b8a027143e16a2f322b5799873/lib/trino-orc/src/main/java/io/trino/orc/OrcReader.java#L471) in `createProjectedLayout`, we should do the same for the field data. Otherwise `createProjectedLayout` is relying on the caller knowing that it needs a downcased list of fields. But I do agree that re-downcasing the list every time you recurse seems inelegant.

I could go either way. @findepi - Can you break the tie?

Reply:
I don't have opinion on this one. @phd3 how hard would it be to do it your way? You can perhaps even push a proposed change to this PR directly, if you wish.

Reply:
@phd3 I moved the downcase to `OrcPageSourceFactory`, let me know if that is what you had in mind. The two tests that failed with the last commit appear to be unrelated to the change, they've have failed and succeeded throughout.